src/components/Purchase/CheckoutPage.js

```diff
@@ -21,34 +22,40 @@ const total = 140.0;
 
 const CheckoutPage = (/*{ album, quantity, subtotal, tax, total}*/) => {
   const [clientSecret, setClientSecret] = useState('');
-
+  const [orderData, setOrderData] = useState(null);
+  const location = useLocation();
   useEffect(() => {
-    // Create PaymentIntent as soon as the page loads
-    const fetchData = async () => {
-      try {
-        const response = await axios.post(
-          'http://localhost:8000/api/v1/orders',
-          {
-            orderItems: [{ album, quantity }],
-            subtotal,
-            tax,
-            total,
-          },
-          {
-            withCredentials: true,
-            headers: { 'Content-Type': 'application/json' },
-          }
-        );
+    // Extract order data from navigation.state
+    const { orderData } = location.state || {};
 
-        setClientSecret(response.data.clientSecret);
-      } catch (error) {
-        // Handle error here
-        console.error('Error fetching data:', error);
-      }
-    };
+    if (orderData) {
+      setOrderData(orderData);
+      // Create PaymentIntent as soon as the page loads
+      const fetchData = async () => {
+        try {
+          const response = await axios.post(
+            `${process.env.REACT_APP_API_BASE_PATH}/orders`,
+            {
+              orderItems: orderData.orderItems, // Use order data received from state
+              subtotal: orderData.subtotal,
+              tax: orderData.tax,
+              total: orderData.total,
```

Comment:
I'm finding that the backend will return a 400 due to Stripe not expecting extra decimals:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/1ecfa349-ce3a-476f-8491-389a776fa053)

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/92c5ef70-51f8-4503-b3c2-aad024571044)

I've put in a quick fix to this in the `main` branch of the backend: Code-the-Dream-School/dd-prac-team4-back@dd82ab4#diff-76a253026a64d07d6b9b82dd96a6d3b4820ed874c4c25e8d0bf4f5b15cbd19e6R30

Reply:
It might still be helpful to only pass in numbers with exactly 2 decimal digits. A function we can use to help do this: https://stackoverflow.com/questions/11832914/how-to-round-to-at-most-2-decimal-places-if-necessary

```javascript
const roundedNumber = (num) => Math.round((num + Number.EPSILON) * 100) / 100
```

Then here we can round all the dollar values (subtotal and total) when passing it to the POST /orders request; and when passing the **total** into the CheckoutForm on line 74 (instead of subtotal)